external_search.py

```diff
@@ -165,18 +164,24 @@ def load(cls, _db, *args, **kwargs):
         return cls(_db, *args, **kwargs)
 
     def __init__(self, _db, url=None, works_index=None, test_search_term=None,
-                 in_testing=False):
+                 in_testing=False, mapping=None):
         """Constructor
 
         :param in_testing: Set this to true if you don't want an
         Elasticsearch client to be created, e.g. because you're
         running a unit test of the constructor.
+
+
+        :param mapping: A custom Mapping object, for use in unit tests. By
+        default, the most recent mapping will be instantiated.
```

Comment:
I haven't done this, but it should be possible to simplify some of our ExternalSearchIndex tests using a very simple Mapping.